actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -23,27 +54,44 @@ module Renderers
     included do
       class_attribute :_renderers
       self._renderers = Set.new.freeze
+      class_attribute :_serializers
+      self._serializers = Hash.new.with_indifferent_access.freeze
     end
 
-    module ClassMethods
-      def use_renderers(*args)
-        renderers = _renderers + args
-        self._renderers = renderers.freeze
-      end
-      alias use_renderer use_renderers
+    def has_serializer?(key)
+      _serializers.key?(key)
+    end
+
+    def serializer_for(key)
+      _serializers[key]
     end
 
+    def assert_serializer!(serializer_name)
+      return true if has_serializer?(serializer_name)
+      raise ActionController::MissingSerializer, "There is no '#{serializer_name}' serializer.\n" <<
+            "Known serializers are #{_serializers.keys}"
+    end
+
+    # Called by +render+ in <tt>AbstractController::Renderering</tt>
+    # which sets the return value as the +response_body+.
+    #
+    # If no renderer is found, +super+ returns control to
+    # <tt>ActionView::Rendering.render_to_body</tt>.
     def render_to_body(options)
       _render_to_body_with_renderer(options) || super
     end
 
     def _render_to_body_with_renderer(options)
       _renderers.each do |name|
-        if options.key?(name)
-          _process_options(options)
-          method_name = Renderers._render_with_renderer_method_name(name)
-          return send(method_name, options.delete(name), options)
-        end
+        next unless options.key?(name)
+        _process_options(options)
+        method_name = Renderers._render_with_renderer_method_name(name)
+        value = options.delete(name)
+        serializer_name = options.key?(:serializer_name) ? options.delete(:serializer_name) : name
+        assert_serializer!(serializer_name)
+        serializer = serializer_for(serializer_name)
+        value = serializer.call(value, options)
+        return send(method_name, value, options)
```

Comment:
TODO: raise ActionController::MissingRenderer instead of ActionView::MissingTemplate